arrow-cast/src/cast.rs

```diff
@@ -527,6 +534,29 @@ fn make_timestamp_array(
     }
 }
 
+fn as_time_res_with_timezone<T: ArrowPrimitiveType>(
+    v: i64,
+    tz: Tz,
+) -> Result<NaiveTime, ArrowError> {
+    match as_datetime_with_timezone::<T>(v, tz) {
+        Some(dt) => Ok(dt.time()),
+        None => Err(ArrowError::CastError(format!(
+            "Failed to create naive time with {} {}",
+            std::any::type_name::<T>(),
+            v
+        ))),
+    }
+}
+
+fn as_timezone_res(tz: &Option<String>) -> Result<Tz, ArrowError> {
+    match tz {
+        Some(s) => s.parse(),
+        None => Err(ArrowError::ParseError(
+            "Timezone must not be none".to_string(),
+        )),
+    }
+}
```

Comment:
We need to support both no timezone or a timezone, perhaps something like (not tested)


```suggestion
fn as_time_res_with_timezone<T: ArrowPrimitiveType>(
    v: i64,
    tz: Option<Tz>,
) -> Result<NaiveTime, ArrowError> {
    let time = match tz {
        Some(tz) => as_datetime_with_timezone::<T>(v, tz).map(|d| d.time()),
        None => as_datetime::<T>(v).map(|d| d.time()),
    };
    
    time.ok_or_else(|| {
        ArrowError::CastError(format!(
            "Failed to create naive time with {} {}",
            std::any::type_name::<T>(),
            v
        ))
    })
}
```

And then parse the timezone with

```
let tz = tz.map(|tz| tz.parse()).transpose()?
```